File: AP2-Big-data-main/AP2-Big-data-main/app/models/matricula.py
from app.database import db
from datetime import datetime

class Matricula(db.Model):
    __tablename__ = 'matriculas'
    
    id = db.Column(db.Integer, primary_key=True, autoincrement=True)
    nome = db.Column(db.String(100), nullable=False)
    email = db.Column(db.String(100), nullable=False)
    curso = db.Column(db.String(100), nullable=False)
    data_matricula = db.Column(db.DateTime, default=datetime.utcnow)
    
    def __init__(self, nome, email, curso):
        self.nome = nome
        self.email = email
        self.curso = curso
    
    def to_dict(self):
        return {
            'id': self.id,
            'nome': self.nome,
            'email': self.email,
            'curso': self.curso,
            'data_matricula': self.data_matricula.isoformat() if self.data_matricula else None
        } 

File: AP2-Big-data-main/AP2-Big-data-main/backend-springboot/src/main/java/com/ibmec/chatbot/entity/Matricula.java
package com.ibmec.chatbot.entity;

import jakarta.persistence.*;
import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Size;
import java.time.LocalDateTime;

@Entity
@Table(name = "matriculas")
public class Matricula {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @NotBlank(message = "Nome é obrigatório")
    @Size(min = 2, max = 100, message = "Nome deve ter entre 2 e 100 caracteres")
    @Column(nullable = false, length = 100)
    private String nome;

    @NotBlank(message = "Email é obrigatório")
    @Email(message = "Email deve ter um formato válido")
    @Size(max = 100, message = "Email deve ter no máximo 100 caracteres")
    @Column(nullable = false, length = 100, unique = true)
    private String email;

    @NotBlank(message = "Curso é obrigatório")
    @Size(min = 2, max = 100, message = "Curso deve ter entre 2 e 100 caracteres")
    @Column(nullable = false, length = 100)
    private String curso;

    @Column(name = "data_matricula", nullable = false)
    private LocalDateTime dataMatricula;

    public Matricula() {
        this.dataMatricula = LocalDateTime.now();
    }

    public Matricula(String nome, String email, String curso) {
        this.nome = nome;
        this.email = email;
        this.curso = curso;
        this.dataMatricula = LocalDateTime.now();
    }

    // Getters e Setters
    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getNome() {
        return nome;
    }

    public void setNome(String nome) {
        this.nome = nome;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public String getCurso() {
        return curso;
    }

    public void setCurso(String curso) {
        this.curso = curso;
    }

    public LocalDateTime getDataMatricula() {
        return dataMatricula;
    }

    public void setDataMatricula(LocalDateTime dataMatricula) {
        this.dataMatricula = dataMatricula;
    }

    @Override
    public String toString() {
        return "Matricula{" +
                "id=" + id +
                ", nome='" + nome + '\'' +
                ", email='" + email + '\'' +
                ", curso='" + curso + '\'' +
                ", dataMatricula=" + dataMatricula +
                '}';
    }
} 